File: web-gui/frontend/src/lib/types/RollerCommand.ts
import { z } from 'zod';

export const RollerCommandSchema = z.object({
    forward: z.number().min(0).max(1).describe("Forward speed for roller drive"),
    backward: z.number().min(0).max(1).describe("Backward speed for roller drive"),
});

export type RollerCommand = z.infer<typeof RollerCommandSchema>;

File: web-gui/backend/ai/classes/RollerCommand.py
from pydantic import BaseModel, Field


class RollerCommand(BaseModel):
    """
    A class to represent a roller command.

    Attributes:
        forward (float): The forward speed for roller drive.
        backward (float): The backward speed for roller drive.
    """

    forward: float = Field(ge=0, le=1, description="Forward speed for roller drive")
    backward: float = Field(ge=0, le=1, description="Backward speed for roller drive")